### PR TITLE
Add transform argument to plots

### LINKED ISSUE #1018: Add transform argument
A transform argument will allow users to pass a function that will be applied to one or more variables. This is related to pymc-devs/pymc3#3420 and pymc-devs/pymc3#3626

### PR DESCRIPTION
**Pull request by amukh18**
## Description
 This pull request adds a much-needed `transform` argument to the function plot_trace. Fixes #1018 

## Checklist

- [x] Does the PR follow [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) 
      PR format?
- [ ] Has included a sample plot to visually illustrate the changes? (only for plot-related functions)
- [ ] Is the new feature properly documented with an example?
- [ ] Does the PR include new or updated tests to cover the new feature (using [pytest fixture pattern](
      https://docs.pytest.org/en/latest/fixture.html#fixture))?
- [x] Is the code style correct (follows pylint and black guidelines)?
- [x] Is the change listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased )?

### CODE REVIEW

**Comment by amukh18** [comment]
@ahartikainen as suggested, I set the default of the `transform` argument to `None`. But I get the error 
`TypeError: 'NoneType' object is not callable`

File: arviz/plots/traceplot.py [hunk 1]
```diff
@@ -137,6 +140,9 @@ def plot_trace(
         divergence_data = False
 
     data = get_coords(convert_to_dataset(data, group="posterior"), coords)
+
+    data = transform(data)
```
Comment (aloctavodia) [inline arviz/plots/traceplot.py]: By doing this you are basically writing `data = None(data)`, which is not legal Python. Instead you should do something like:

```python
if transform is not None:
    data = transform(data)
```

File: arviz/plots/traceplot.py [hunk 2]
```diff
@@ -46,6 +47,8 @@ def plot_trace(
         Coordinates of var_names to be plotted. Passed to `Dataset.sel`
     divergences : {"bottom", "top", None, False}
         Plot location of divergences on the traceplots. Options are "bottom", "top", or False-y.
+    transform : callable
+    	Function to transform data (defaults to identity)
```
Comment (aloctavodia) [inline arviz/plots/traceplot.py]: Maybe a better wording will be "Function to transform data (defaults to None i.e. the identity function)" or probably just "Function to transform data (defaults to None)"

**Comment by aloctavodia** [comment]
@amukh18 Also notices that `plot_trace` is not the only function needing this new argument. Other functions are `plot_forest, `plot_pair`, `plot_posterior`, `plot_rank`, `plot_parallel`,  `plot_violin`, `plot_density`, `plot_joint`.

**Comment by aloctavodia** [comment]
@amukh18 any update on this?

**Comment by amukh18** [comment]
@aloctavodia  I deeply apologise for the long delay but I was caught up in my coursework and exams. I am relatively free now and have made the necessary commits.

File: arviz/plots/traceplot.py (hunk 1 above)
Comment (amukh18) [reply arviz/plots/traceplot.py]: @aloctavodia Sorry for this elementary mistake! I fixed it.

**Comment by amukh18** [comment]
I have spotted some errors. I will recommit

**Comment by amukh18** [comment]
My code passed local pylint checks but the Travis CI build failed

**Comment by amukh18** [comment]
@OriolAbril No problem!
And yes, I'll add those examples

**Comment by amukh18** [comment]
@OriolAbril I have created the pull request and am working on it.

**Comment by OriolAbril** [comment]
I am merging this so examples can be added on top of this using `transform` already

**Pull request merged**